crates/wit-parser/src/lib.rs

```diff
@@ -419,6 +432,8 @@ pub struct Interface {
     /// This is `None` for inline interfaces in worlds.
     pub name: Option<String>,
 
+    /// The nested interfaces in this interface.
+    pub nested: IndexMap<String, Nest>,
```

Comment:
One thing that might be helpful here (and we should probably do elsewhere but is fine to leave for a separate PR) is to do a `skip_serializing_if` if the map is empty. That'll help reduce the diff in the test files in this PR and I think it makes sense in general to skip serializing empty things too.

Reply:
Also, thinking more on this, might this perhaps make sense as a `Vec<Nest>` instead of an `IndexMap`? The use case for looking up a specific nest item I think is a little less than it is with functions/types. Another possible key for this map could be `InterfaceId` instead of `String` since the string should be able to be inferred from the id. (that may not play well with json though perhaps...)